library/pod-security-policy/allow-privilege-escalation/src.rego

```diff
@@ -0,0 +1,25 @@
+package k8spspallowprivilegeescalationcontainer
+
+violation[{"msg": msg, "details": {}}] {
+    c := input_containers[_]
+    input_allow_priviledge_escalation_allowed(c)
+    msg := sprintf("Privilege escalation container is not allowed: %v", [c.name])
+}
+
+input_allow_priviledge_escalation_allowed(c) {
+    not has_field(c, "securityContext")
+}
+input_allow_priviledge_escalation_allowed(c) {
+    has_field(c, "securityContext")
+    has_field(c.securityContext, "allowPrivilegeEscalation")
+}
```

Comment:
This is written very clearly.  Alternatively, you could write 

input_allow_priviledge_escdalation_allowed(c) {
    c.securityContext.allowPrivilegeEscalation
}

Rego turns missing fields into undefined.